src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,23 @@
+package duke.task;
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public String writeToFile() {
+        String done="0";
+        if (super.getDone()) done="1";
```

Comment:
Do note that it is good practice to properly enclose the single statement with curly braces, to avoid future bugs when you may want to expand this to multiple statements. This goes for the same statement in `ToDo.java` and `Event.java` as well.